lang/cs/Org.Apache.REEF.Common/metrics/MetricsSystem/MetricsRecordBuilder.cs

```diff
@@ -0,0 +1,186 @@
+        /// <returns>Self reference.</returns>
+        public IMetricsRecordBuilder Add(IImmutableMetric metric)
+        {
+            _metrics.Add(metric);
+            return this;
+        }
+
+        /// <summary>
+        /// Sets a context for the record builder. Creates the corresponding tag.
+        /// </summary>
+        /// <param name="value">Value of the context.</param>
+        /// <returns>Self reference.</returns>
+        public IMetricsRecordBuilder SetContext(string value)
+        {
+            _contextValue = value;
+            return AddTag(MetricsSystemConstants.Context, value);
+        }
+
+        /// <summary>
+        /// Adds a counter to the record builder.
+        /// </summary>
+        /// <param name="info">Meta data info of the counter.</param>
+        /// <param name="value">Value of the counter.</param>
+        /// <returns>Self reference.</returns>
+        public IMetricsRecordBuilder AddCounter(IMetricsInfo info, long value)
+        {
+            _metrics.Add(new ImmutableCounter(info, value));
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a long gauge to the record builder.
+        /// </summary>
+        /// <param name="info">Meta data info of the gauge.</param>
+        /// <param name="value">Value of the gauge.</param>
+        /// <returns>Self reference.</returns>
+        public IMetricsRecordBuilder AddGauge(IMetricsInfo info, long value)
+        {
+            _metrics.Add(new ImmutableLongGauge(info, value));
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a double gauge to the record builder.
+        /// </summary>
+        /// <param name="info">Meta data info of the gauge.</param>
+        /// <param name="value">Value of the gauge.</param>
+        /// <returns>Self reference.</returns>
+        public IMetricsRecordBuilder AddGauge(IMetricsInfo info, double value)
+        {
+            _metrics.Add(new ImmutableDoubleGauge(info, value));
+            return this;
+        }
+
+        /// <summary>
+        /// Returns the parent collector.
+        /// </summary>
+        public IMetricsCollector ParentCollector()
+        {
+            return _parentCollector;
+        }
+
+        /// <summary>
+        /// Ends the record building and returns the parent collector.
+        /// </summary>
+        /// <returns>The parent collector.</returns>
+        public IMetricsCollector EndRecord()
+        {
+            return _parentCollector;
+        }
+
+        /// <summary>
+        /// Creates the record from record builder.
+        /// </summary>
+        /// <returns>The newly created record. Return null if no metric or tag exists.</returns>
+        public IMetricsRecord GetRecord()
+        {
+            TimeSpan t = DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1));
+            long timeStamp = (long)t.TotalSeconds;
+
+            if (_metrics.Count != 0 || _tags.Count != 0)
```

Comment:
Please use `IsEmpty()` instead or checking the condition explicitly, in case the condition changes.